nmostesting/GenericTest.py

```diff
@@ -383,6 +397,97 @@ def validate_schema(self, payload, schema):
     def do_request(self, method, url, **kwargs):
         return TestHelper.do_request(method=method, url=url, **kwargs)
 
+    def make_key_cert_files(self, cert_file, key_file):
+        # create a key pair
+        k = crypto.PKey()
+        k.generate_key(crypto.TYPE_RSA, 2048)
+
+        # create cert
+        cert = crypto.X509()
+        cert.set_version(2)
+        cert.get_subject().C = "GB"
+        cert.get_subject().ST = "England"
+        cert.get_subject().O = "NMOS Testing Ltd"  # noqa: E741
+        ca_cert_subject = cert.get_subject()
+        ca_cert_subject.CN = "ca.testsuite.nmos.tv"
+        cert.set_issuer(ca_cert_subject)
+        cert.get_subject().CN = "mocks.testsuite.nmos.tv"
+        cert.set_serial_number(x509.random_serial_number())
+        cert.gmtime_adj_notBefore(0)
+        cert.gmtime_adj_notAfter(10*365*24*60*60)
+        cert.set_pubkey(k)
+        # get Root CA key
+        capkey = open(CONFIG.KEY_TRUST_ROOT_CA, "r").read()
+        ca_pkey = crypto.load_privatekey(crypto.FILETYPE_PEM, capkey)
+        # get Root CA cert
+        cacert = open(CONFIG.CERT_TRUST_ROOT_CA, "r").read()
+        ca_cert = crypto.load_certificate(crypto.FILETYPE_PEM, cacert)
+        # create cert extension
+        san = ["DNS:mocks.{}".format(CONFIG.DNS_DOMAIN), "DNS: nmos-mocks.local"]
+        cert_ext = []
+        cert_ext.append(crypto.X509Extension(b'subjectKeyIdentifier', False, b'hash', cert))
+        cert_ext.append(crypto.X509Extension(b'authorityKeyIdentifier', False, b'keyid,issuer:always', issuer=ca_cert))
+        cert_ext.append(crypto.X509Extension(b'basicConstraints', False, b'CA:FALSE'))
+        cert_ext.append(crypto.X509Extension(b'keyUsage', True, b'digitalSignature, keyEncipherment'))
+        cert_ext.append(crypto.X509Extension(b'subjectAltName', False, ','.join(san).encode()))
+        cert.add_extensions(cert_ext)
+
+        # sign cert with Intermediate CA key
+        cert.sign(ca_pkey, 'sha256')
+
+        # write chain certificate file
+        if cert_file is not None:
+            with open(cert_file, "wt") as f:
```

Comment:
This reminds me, should we be documenting somewhere that a writeable filesystem is required... or not writing anything to file, just keeping it in memory?

Reply:
I have that thought too, but not found an equivalent function of loading the memory into the ctx rather than using load_cert_chain file based.

Reply:
Right... same conclusion as https://github.com/AMWA-TV/nmos-testing/pull/599#discussion_r1094321233